### PR TITLE
More robust af calc

### PR DESCRIPTION
**Pull request by shannog2**
# Summary

- The calculation of AF for annotated files has been made more robust to different vcf file formats. 

- "Coverage" is now also added to annotated files using the DP field

- Pairwise vcf comparisons for single variants has been sped up by restricting the vcf being searched to variants that are on the same chromosome as the single variant

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -253,6 +253,22 @@ def write_vcf(lines, vcf):
     return vcf
 
 
+def write_filtered_vcf(vcf, chrm, out_vcf):
+
+    content = []
+
+    with versatile_open(vcf, "r") as vcf_handle:
+
+        for line in vcf_handle.readlines():
```
Comment (marghoob) [inline utils.py]: Is there a reason for loading the VCF in memory? Why not simply `for line in vcf_handle:`?

Comment (shannog2) [reply utils.py]: Good spot, I'll change this

File: utils.py [hunk 2]
```diff
@@ -304,7 +320,9 @@ def make_clean_vcf(vcf, path=None):
                 continue
             else:
                 GT = "0/1" if line_split[-1] == "./." else line_split[-1]
-                clean_vcf_handle.write('\t'.join([line_split[0], line_split[1], ".", line_split[3], line_split[4], ".", ".", ".", line_split[8], GT]) + '\n')
+                info_entries = [line_split[7].split('=')[0] for x in line_split[7].split(';')]
```
Comment (marghoob) [inline utils.py]: What is this capturing? The iterated variable `x` does not seem to be used here.

Comment (shannog2) [reply utils.py]: Good spot, that was a bug. In very few cases would it have any negative effect which is why it wasn't caught in tests

File: compare_vcf.py [hunk 3]
```diff
@@ -356,24 +363,38 @@ def match_false(augmented_file, files_to_pair_with, out_dir, sample, log_to_file
...
+                        AO_RO_DP = {"AO": None, "RO": None, "DP": None}
                         if equivalent_variant:
-                            try:
-                                AO = int(equivalent_variant[-1].split(':')[4].split(',')[0])
-                                RO = int(equivalent_variant[-1].split(':')[2].split(',')[0])
-                            except:
-                                info = "N/A;"
-                            else:
-                                info = str(float(AO)/(AO+RO)) + ';'
+                            for entry in AO_RO_DP:
```
Comment (yunfeiguo) [inline compare_vcf.py]: Hi @shannog2 could you wrap this part into a (preferably generic) function?

**Comment by yunfeiguo** [comment]
Sorry @marghoob I accidentally clicked the re-requesting review button.

File: compare_vcf.py (hunk 3 above)
Comment (shannog2) [reply compare_vcf.py]: Done

File: compare_vcf.py [hunk 4]
```diff
@@ -369,32 +369,25 @@ def match_false(augmented_file, files_to_pair_with, out_dir, sample, log_to_file
...
-                                    pass
-                        if AO_RO_DP["AO"] and AO_RO_DP["RO"]:
-                            info = str(float(AO_RO_DP["AO"])/(AO_RO_DP["AO"]+AO_RO_DP["RO"])) + ';'
+                            for entry in AO_RO_DP_AD:
+                                AO_RO_DP_AD[entry] = utils.get_info(equivalent_variant, entry)
+
+                        # gatk4 format
+                        if AO_RO_DP_AD["AD"]:
+                            AD_split = AO_RO_DP_AD["AD"].split(',')
+                            AO = int(AD_split[1])
```
Comment (yunfeiguo) [inline compare_vcf.py]: There could be more than 1 AO number and multiple ALT alleles, we need to take care of that.

Comment (shannog2) [reply compare_vcf.py]: Done

**Comment by yunfeiguo** [comment]
Feel free to merge when you are comfortable.

**Pull request merged**